share/p2p/peers/manager.go

```diff
@@ -482,12 +497,24 @@ func (m *Manager) cleanUp() []peer.ID {
 
 	addToBlackList := make(map[peer.ID]struct{})
 	for h, p := range m.pools {
-		if !p.isValidatedDataHash.Load() && time.Since(p.createdAt) > m.params.PoolValidationTimeout {
-			delete(m.pools, h)
-			if p.headerHeight.Load() < m.initialHeight.Load() {
-				// outdated pools could still be valid even if not validated, no need to blacklist
-				continue
+		if p.isValidatedDataHash.Load() {
+			// remove pools that are outdated
+			if p.height < uint64(m.storeFrom.Load()) {
+				delete(m.pools, h)
 			}
+			continue
+		}
+
+		// can't validate datahashes below initial height
+		if p.height < m.initialHeight.Load() {
+			delete(m.pools, h)
+			continue
+		}
+
+		// find pools that are not validated in time
+		if time.Since(p.createdAt) > m.params.PoolValidationTimeout {
```

Comment:
I think the validation timeout should be defined in heights now:
* Removes the need for createdAt (time operations are costly)
* In general, it is easier to reason about if we make `storedPoolsAmount` with heights.

We should also decrease it for LNs. I don't see a point of keeping pools for 2mins for LNs

Reply:
Decided in sync to do defer this, s.t. we can release the fix asap.

Reply:
As we discussed privately. Can't rely on incoming height, as it could be set by sender to any value higher than know network head. It creates situation, when it is diffucult to distinguish if msg recieved was for invalid datahash, or headerSub is lagging behind shrexSub.
There could be simplier approach, that would require some refactoring, so I'll land it into separate issue.

Reply:
https://github.com/celestiaorg/celestia-node/issues/3011